### PR TITLE
Add `argocd`

### PR DESCRIPTION
**Pull request by dhirschfeld**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/argocd-cli/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  url:  https://github.com/argoproj/argo-cd/archive/v{{ version }}.tar.gz
+  sha256: b8ea6665bc37a34de0a6fe7592fb8ae376847e1c93fc5d6377140a98c1aa6a55
+
+build:
+  skip: True  # [win]
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('go') }}
```
Comment (xhochy) [inline recipes/argocd-cli/meta.yaml:18]:
```suggestion
    - {{ compiler('go-ncgo') }}
```

File: recipes/argocd-cli/build.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -exuo pipefail
+
+export CGO_ENABLED=0
```
Comment (xhochy) [inline recipes/argocd-cli/build.sh:4]:
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by dhirschfeld** [comment]
Thanks for the review @xhochy! 

I assume by using `go-ncgo` I don't need the `CGO_ENABLED=0` env var?

If so, I'll make the same change to my other PR #27044

**Comment by dhirschfeld** [comment]
Looks like it doesn't like `go-ncgo`?
```
 - nothing provides requested go-ncgo_linux-64
```

**Comment by dhirschfeld** [comment]
I'll see if `go` works whilst trying to figure out what was happening with `go-ncgo`...

File: recipes/argocd-cli/meta.yaml (hunk 1 above)
Comment (xhochy) [reply recipes/argocd-cli/meta.yaml:18]: Typo. It should be go-nocgo

**Comment by dhirschfeld** [comment]
Ping @conda-forge/staged-recipes, @conda-forge/go - this should be good to review/merge!

**Comment by dhirschfeld** [comment]
Added `go-nocgo` to `kind` (#27044) as well.

File: recipes/argocd-cli/meta.yaml (hunk 1 above)
Comment (dhirschfeld) [reply recipes/argocd-cli/meta.yaml]: updated!

**Review by xhochy — APPROVED** [review]

**Pull request merged**